tests/std/tests/P2165R4_tuple_like_tuple_members/test.cpp

```diff
@@ -0,0 +1,280 @@
+
+        Tuple<const TmpChar&> t5(as_const(a));
+        assert((&get<Indices>(t5) == &get<Indices>(a)) && ...);
+
+        Tuple<TmpChar> t6(std::move(a));
+        assert((get<Indices>(t6) == get_letter(Indices) && get<Indices>(a) == '\x7F') && ...);
+        ((get<Indices>(a) = std::move(get<Indices>(t6))), ...);
+
+        Tuple<const TmpChar&> p7(std::move(a));
+        assert((&get<Indices>(p7) == &get<Indices>(a)) && ...);
+
+        Tuple<TmpChar&&> t8(std::move(a));
+        assert((&get<Indices>(t8) == &get<Indices>(a)) && ...);
+
+        Tuple<const TmpChar&&> t9(std::move(a));
+        assert((&get<Indices>(t9) == &get<Indices>(a)) && ...);
+
+        Tuple<TmpChar> t10(std::move(as_const(a)));
+        assert((get<Indices>(t10) == get_letter(Indices) && get<Indices>(a) == get_letter(Indices)) && ...);
+
+        Tuple<const TmpChar&> t11(std::move(as_const(a)));
+        assert((get<Indices>(t11) == get_letter(Indices) && get<Indices>(a) == get_letter(Indices)) && ...);
+
+        Tuple<const TmpChar&&> t12(std::move(as_const(a)));
+        assert((get<Indices>(t12) == get_letter(Indices) && get<Indices>(a) == get_letter(Indices)) && ...);
+
+        // Check explicit specifier
+        static_assert(!convertible_to<string_view, string>);
+        static_assert(convertible_to<TupleLike<int>, Tuple<long>>);
+        static_assert(!convertible_to<TupleLike<string_view>, Tuple<string>> || N == 0);
+        static_assert(constructible_from<Tuple<string>, TupleLike<string_view>>);
+
+        return true;
+    }
+
+    template <size_t... Indices>
+    static constexpr bool test_tuple_like_allocator_constructor(index_sequence<Indices...>) {
+        // Test tuple(tuple-like, allocator_arg_t, Alloc)
+        static_assert(constructible_from<Tuple<string>, allocator_arg_t, allocator<char>, TupleLike<string>&>);
+        static_assert(constructible_from<Tuple<string>, allocator_arg_t, allocator<char>, const TupleLike<string>&>);
+        static_assert(constructible_from<Tuple<string>, allocator_arg_t, allocator<char>, TupleLike<string>>);
+        static_assert(constructible_from<Tuple<string>, allocator_arg_t, allocator<char>, const TupleLike<string>>);
+
+        if (!is_constant_evaluated()) {
+            using Alloc = pmr::polymorphic_allocator<char>;
+
+            array<byte, 128> buffer;
+            pmr::monotonic_buffer_resource resource{buffer.data(), buffer.size()};
+            auto get_str = [&](size_t Idx) { return pmr::string({static_cast<char>('A' + Idx)}, Alloc{&resource}); };
+
+            TupleLike<pmr::string> a = {get_str(Indices)...}; // "A"s, "B"s, "C"s, ...
+
+            Tuple<pmr::string> t1(allocator_arg, Alloc{&resource}, a);
+            assert((get<Indices>(t1).get_allocator() == get<Indices>(a).get_allocator()) && ...);
+            assert((ranges::equal(get<Indices>(t1), get_str(Indices)) && //
+                       ranges::equal(get<Indices>(a), get_str(Indices)))

```

Comment:
Nitpick: We conventionally place binary operators at the beginning of lines (under the belief that they are easier to notice there). Occurs below.